### DISCUSSION TITLE
Reply to SYN package

### DISCUSSION THREAD

**Discussion by damjanmi** [scottoffen/grapevine#127] [Q&A] [answered]
Hi,

Can grapevine reply to SYN package with ACK? I have a client that is first probing server if it's alive and then sending post to endpoint.

Damjan

**Comment by scottoffen** [comment, marked as answer]
SYN and ACK packages are used for initializing TCP connection. Grapevine works on the HTTP layer. The underlying TCP connection is completely obfuscated from Grapevine via its use of [`HttpListener`](https://docs.microsoft.com/en-us/dotnet/api/system.net.httplistener?view=net-6.0), so it cannot be configured to respond to a SYN request.
